healthcheck/generalchecks/src/main/java/org/apache/felix/hc/generalchecks/BundlesStartedCheck.java

```diff
@@ -79,78 +82,85 @@ protected void activate(BundleContext bundleContext, Config config) {
+        log.info("Bundles total: {}, expected to be active: {}, active: {}.", bundles.size(), checkedBundles, activeBundles);
+        return new Result(log);
+    }
 
-        Bundle[] bundles = this.bundleContext.getBundles();
-        log.debug("Framwork has {} bundles in total", bundles.length);
-
-        int countExcluded = 0;
-        int relevantBundlesCount = 0;
-        int inactiveCount = 0;
-        for (Bundle bundle : bundles) {
-            String bundleSymbolicName = bundle.getSymbolicName();
-            int bundleState = bundle.getState();
-
-            if (!includesRegex.matcher(bundleSymbolicName).matches()) {
-                LOG.debug("Bundle {} not matched by {}", bundleSymbolicName, includesRegex);
-                continue;
-            }
-
-            if (excludesRegex.isPresent() && excludesRegex.get().matcher(bundleSymbolicName).matches()) {
-                LOG.debug("Bundle {} excluded {}", bundleSymbolicName, excludesRegex);
-                countExcluded ++;
-                continue;
-            }
-            relevantBundlesCount++;
-
-            boolean bundleIsLogged = false;
-            if (bundleState != Bundle.ACTIVE) {
-                // support lazy activation (https://www.osgi.org/developer/design/lazy-start/)
-                if (bundleState == Bundle.STARTING && isLazyActivation(bundle)) {
-                    LOG.debug("Ignoring lazily activated bundle {}", bundleSymbolicName);
-                } else  if (StringUtils.isNotBlank(bundle.getHeaders().get(Constants.FRAGMENT_HOST))) {
-                    LOG.debug("Ignoring bundle fragment: {}", bundleSymbolicName);
-                } else {
-                    String msg = "Inactive bundle {} {}: {}";
-                    Object[] msgObjs = new Object[] {bundle.getBundleId(), bundleSymbolicName, getStateLabel(bundleState)};
-                    LOG.debug(msg, msgObjs);
-                    if(useCriticalForInactive) {
-                        log.critical(msg, msgObjs);
-                    } else {
-                        log.warn(msg, msgObjs);
-                    }
-                    bundleIsLogged = true;
-                    inactiveCount++;
-                }
-            }
-            if(!bundleIsLogged) {
-                log.debug("Bundle {} {}: {}", bundle.getBundleId(), bundleSymbolicName, getStateLabel(bundleState));
-            }
+    private boolean isNotFragment(Bundle bundle) {
+        boolean isFragment = StringUtils.isNotBlank(bundle.getHeaders().get(Constants.FRAGMENT_HOST));
+        if (isFragment) {
+            LOG.debug("Ignoring bundle fragment: {}", bundle.getSymbolicName());
         }
-
-        String includeMsg = !includesRegex.pattern().equals(".*") ? " for pattern " + includesRegex.pattern(): "";
-        String baseMsg = relevantBundlesCount + " bundles" + includeMsg;
-        String excludedMsg = countExcluded > 0 ? " (" + countExcluded + " excluded via pattern " + excludesRegex.get().pattern() + ")" : "";
```

Comment:
Showing the exclude pattern in the result for the case it kicks in is quite useful IMHO - with this PR I only see `Bundles total: 578, expected to be active: 567, active: 566` and I don't immediately see the reason why 567 out of 578 are expected (someone might think this is due to the number 567 being configured when in reality it is an exclude regex)